packages/x-date-pickers/src/internals/components/FakeTextField/FakeTextField.tsx

```diff
@@ -1,68 +1,124 @@
 import * as React from 'react';
-import Box from '@mui/material/Box';
+import clsx from 'clsx';
+import { styled } from '@mui/material/styles';
+import { unstable_composeClasses as composeClasses, unstable_useId as useId } from '@mui/utils';
+import InputLabel from '@mui/material/InputLabel';
+import FormHelperText from '@mui/material/FormHelperText';
+import FormControl from '@mui/material/FormControl';
+import { getFakeTextFieldUtilityClass } from './fakeTextFieldClasses';
+import FakeInput from './FakeInput';
+import { FakeTextFieldProps } from './FakeTextField.types';
 
-export interface FakeTextFieldElement {
-  container: React.HTMLAttributes<HTMLSpanElement>;
-  content: React.HTMLAttributes<HTMLSpanElement>;
-  before: React.HTMLAttributes<HTMLSpanElement>;
-  after: React.HTMLAttributes<HTMLSpanElement>;
-}
+const FakeTextFieldRoot = styled(FormControl, {
+  name: 'MuiFakeTextField',
+  slot: 'Root',
+  overridesResolver: (props, styles) => styles.root,
+})({});
 
-interface FakeTextFieldProps {
-  elements: FakeTextFieldElement[];
-  valueStr: string;
-  onValueStrChange: React.ChangeEventHandler<HTMLInputElement>;
-  error: boolean;
-  id?: string;
-  InputProps: any;
-  inputProps: any;
-  disabled?: boolean;
-  autoFocus?: boolean;
-  ownerState?: any;
-  valueType: 'value' | 'placeholder';
-}
+const useUtilityClasses = (ownerState: FakeTextFieldProps) => {
+  const { focused, disabled, classes, required } = ownerState;
+
+  const slots = {
+    root: [
+      'root',
+      focused && !disabled && 'focused',
+      disabled && 'disabled',
+      required && 'required',
+    ],
+  };
+
+  return composeClasses(slots, getFakeTextFieldUtilityClass, classes);
+};
 
 export const FakeTextField = React.forwardRef(function FakeTextField(
   props: FakeTextFieldProps,
   ref: React.Ref<HTMLDivElement>,
 ) {
   const {
     elements,
+    className,
+    color = 'primary',
+    disabled = false,
+    error = false,
+    label,
+    variant = 'outlined',
+    fullWidth = false,
     valueStr,
-    onValueStrChange,
-    id,
-    error,
-    InputProps,
-    inputProps,
-    autoFocus,
-    disabled,
+    helperText,
     valueType,
-    ownerState,
+    id: idOverride,
+    FormHelperTextProps,
+    InputLabelProps,
+    inputProps,
+    InputProps,
+    required = false,
     ...other
   } = props;
 
+  const [focused, setFocused] = React.useState(false);
+
+  const areAllSectionsEmpty = !valueStr && valueType !== 'placeholder';
```

Comment:
`valueStr` should never be empty, even when all sections are empty we still have the placeholder (`MM/DD/YYYY` for instance).
Not sure to understand the logic here.

I'm crearting `valueType` this way:

```tsx
valueType: isFocused && areAllSectionsEmpty ? 'placeholder' : 'value',
```

But if you have to re-create `areAllSectionsEmpty` here, then we should probably return directly `areAllSectionsEmpty` (maybe rename it `isEqualToEmptyValue` since it's what the variable truly means).